lib/sudo-exec/src/monitor.rs

```diff
@@ -115,6 +120,46 @@ impl MonitorClosure {
 
         false
     }
+
+    /// Based on `mon_backchannel_cb`
+    fn read_backchannel(&mut self, event_handler: &mut EventHandler<Self>) {
+        match self.backchannel.recv() {
+            // Read interrupted, we can try again later.
+            Err(err) if was_interrupted(&err) => {}
+            // There's something wrong with the backchannel, break the event loop
+            Err(err) => {
+                event_handler.set_break();
+                self.backchannel.send((&err).into()).unwrap();
+            }
+            Ok(event) => {
+                match event {
+                    // We shouldn't receive this event more than once.
+                    MonitorEvent::ExecCommand => unreachable!(),
+                    // Forward signal to the command.
+                    MonitorEvent::Signal(signal) => self.send_signal(signal),
+                }
+            }
+        }
+    }
+
+    /// Send a signal to the command
+    fn send_signal(&self, signal: c_int) {
+        // FIXME: We should call `killpg` instead of `kill`.
+        // FIXME: We shouldn't send any signals if the command exited already.
```

Comment:
can you change the `command_pid` to an Option and set it to `None` when `command.try_wait` succeeds? this would then be executed conditionally: only when `command_pid` is in the `Some` variant

Reply:
Yeah that's exactly what ogsudo does :sweat_smile: (but using `-1` :P).